python.d/w1sensor.chart.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+# Description: 1-wire temperature monitor netdata python.d module
+# Author: Diomidis Spinellis <http://www.spinellis.gr>
+
+import os
+import re
+from bases.FrameworkServices.SimpleService import SimpleService
+
+# default module values (can be overridden per job in `config`)
+update_every = 5
+priority = 60000
+retries = 60
+
+# Location where 1-Wire devices can be found
+W1_DIR = '/sys/bus/w1/devices/'
+
+# Lines matching the following regular expression contain a temperature value
+RE_TEMP = re.compile(r' t=(\d+)')
+
+ORDER = ['temp']
+
+# id: {
+#     'options': [name, title, units, family (heading 2), context, charttype],
+#     'lines': [
+#         [unique_dimension_name, name, algorithm, multiplier, divisor]
+#     ]}
+CHARTS = {
+    'temp': {
+        'options': [
+            None,
+            '1-Wire Temperature Sensor',
+            'Celsius',
+            'Temperature',
+            'w1sensor.temp',
+            'line'
+        ],
+        # lines are constructed from check
+    }
+}
+
+# Known and supported family members
+# Based on linux/drivers/w1/w1_family.h and w1/slaves/w1_therm.c
+THERM_FAMILY = {
+    '10': 'W1_THERM_DS18S20',
+    '22': 'W1_THERM_DS1822',
+    '28': 'W1_THERM_DS18B20',
+    '3b': 'W1_THERM_DS1825',
+    '42': 'W1_THERM_DS28EA00',
+}
+
+class Service(SimpleService):
+    """Provide netdata service for 1-Wire sensors"""
+    def __init__(self, configuration=None, name=None):
+        SimpleService.__init__(self, configuration=configuration, name=name)
+        self.order = ORDER
+        self.definitions = CHARTS
+        self.probes = []
+
+    def check(self):
+        """Locate available 1-Wire sensors, setting definitions and probes."""
+        lines = []
+        try:
+            # Auto-detect available 1-Wire temperature sensors
+            for file_name in os.listdir(W1_DIR):
+                if file_name[2] != '-':
+                    continue
+                if not file_name[0:2] in THERM_FAMILY:
+                    continue
+                self.probes.append(file_name)
+
+                identifier = file_name[3:]
+                name = identifier
+                config_name = self.configuration.get('name_' + identifier)
+                if config_name:
+                    name = config_name
+                lines.append(['w1sensor_temp_' + identifier, name,
+                              'absolute', 1, 10])
+            self.definitions['temp']['lines'] = lines
+            return len(self.probes) > 0
+        except Exception as error:
```

Comment:
Some suggestions:
1. Make `exception` more specific
2. Remove all code from `try/except` block which can not cause an exception

Like:

```python
try:
    filenames = os.listdir(W1_DIR)
except OSError as err:  # i believe we only need OSError here, but i am not sure
    self.error(err)
    return None

...
...
```

Reply:
Done.